packages/lisk-transactions/src/utils/validation/validation.ts

```diff
@@ -83,14 +84,25 @@ export const validateAddress = (address: string) => {
 		);
 	}
 
-	const addressAsBignum = new BigNum(address.slice(0, -1));
+	if (address.indexOf('.') > -1) {
+		throw new Error(
+			'Address format does not match requirements. Address includes invalid character: `.`.',
+		);
+	}
 
-	if (addressAsBignum.cmp(new BigNum(MAX_ADDRESS_NUMBER)) > 0) {
+	const addressString = address.slice(0, -1);
+	const addressNumber = new BigNum(addressString);
+
+	if (addressNumber.cmp(new BigNum(MAX_ADDRESS_NUMBER)) > 0) {
 		throw new Error(
 			'Address format does not match requirements. Address out of maximum range.',
 		);
 	}
 
+	if (addressString !== addressNumber.toString(BASE_TEN)) {
+		throw new Error('Address number does not have natural representation.');
```

Comment:
I'm not sure about the "natural representation".
Maybe better to say directly that string format doesn't match the number?

Reply:
This is the wording used in core, but I agree not very clear.